records/2022-Q4-activity-report.md

```diff
@@ -0,0 +1,420 @@
+
+### Language Improvements
+
+For Scala 3.
+
+#### Clause Interleaving
+
+As a reminder, last quarter, [SIP-47 - Clause Interleaving](https://github.com/scala/improvement-proposals/blob/main/content/clause-interleaving.md) proposed by the Center, got accepted for implementation.
+As described in the Q3 report, it provides a way to have type parameter bounds be dependent on value parameters.
+
+During this quarter, work has continued on bringing this SIP into the compiler as an experimental feature.
+
+Particular attention has been ported to the [pull request](https://github.com/lampepfl/dotty/pull/14019) and commit messages, to make them as easy to read and understand as possible.
+This way it will be usable as a guide when writing future SIP implementations.
+
+The code itself also follows this principle, bringing well documented precise changes that make the project more simple, not less.
+
+#### Case Classes that Can Evolve in a Binary-Compatible Way
+
+Case classes are popular to achieve domain modeling. However, they
+come with an important drawback: it is impossible to apply any evolution
+(adding/removing fields on a case class) in a backward binary compatible way.
+This forces library authors to manually re-implement the features of
+case classes that they need (typically, structural equality) at the price of
+more verbosity. Other solutions include tools generating code (e.g.
+contraband), or experimental macro annotations (e.g. data-class, scalameta).
+
+In Q4, we submitted a [proposal](https://github.com/scala/improvement-proposals/pull/50) that has been actively
+discussed by the Committee. Ultimately, we decided to withdraw the proposal in favor of documenting a workaround
+that supports our use case without changing the language.
+
+### TASTy Manipulation Library (tasty-query)
+
+For Scala 3.
+
+tasty-query is a library to read semantic information from Scala 3 classpaths.
+It reads Scala .tasty files, Scala 2 pickles and Java .class files, and presents the semantic information they contain in a unified API.
+
+tasty-query will be an essential tool for any static analysis involving Scala code.
+In particular, it is the basis for TASTy-MiMa.
+More information can be found [in the readme of tasty-query](https://github.com/scalacenter/tasty-query#readme).
+
+In this quarter, we made significant progress on tasty-query, and released several in-progress versions, the latest being 0.5.5.
+tasty-query is already used in the Scala Debug Adapter to provide smart step-into filters for Scala 3 code, as well as in the in-progress TASTy-MiMa.
+
+### TASTy-MiMa
+
+For Scala 3.
+
+TASTy-MiMa is a new, in-progress tool that checks TASTy-compatibility across library versions.
+It is similar to MiMa, which works at the binary compatibility level.
```

Comment:
Let's link to MiMa here in case people aren't familiar with it.

Reply:
Done in fce926e36f94706c5bedd331b59ffca522fecc59